contracts/gelato_executor_modules/PermissionedExecutors.sol

```diff
@@ -0,0 +1,137 @@
+// "SPDX-License-Identifier: UNLICENSED"
+pragma solidity ^0.6.10;
+pragma experimental ABIEncoderV2;
+
+import {IGelatoCore, TaskReceipt} from "../gelato_core/interfaces/IGelatoCore.sol";
+import {IGelatoProviders} from "../gelato_core/interfaces/IGelatoProviders.sol";
+import {IGelatoExecutors} from "../gelato_core/interfaces/IGelatoExecutors.sol";
+import {IGelatoSysAdmin} from "../gelato_core/interfaces/IGelatoSysAdmin.sol";
+import {Address} from  "../external/Address.sol";
+import {GelatoTaskReceipt} from "../libraries/GelatoTaskReceipt.sol";
+import {GelatoString} from "../libraries/GelatoString.sol";
+
+/// @title PermissionedExecutors
+/// @notice Contract that masks 2 known executor addresses behind one address
+/// @author gitpusha & HilmarX
+contract PermissionedExecutors {
+
+    using Address for address payable;
+    using GelatoTaskReceipt for TaskReceipt;
+    using GelatoString for string;
+
+    struct Reponse {
+        uint256 taskReceiptId;
+        uint256 taskGasLimit;
+        string response;
+    }
+
+    address public immutable gelatoCore;
+    // First Executor == Luis Dev Index 1
+    address public constant first_executor = 0x4d671CD743027fB5Af1b2D2a3ccbafA97b5B1B80;
+    // Second Executor == Hilmar Index 1
+    address public constant second_executor = 0x99E69499973484a96639f4Fb17893BC96000b3b8;
+
+    constructor(address _gelatoCore) public payable {
+        gelatoCore = _gelatoCore;
+        if (msg.value >= IGelatoSysAdmin(_gelatoCore).minExecutorStake())
+            IGelatoExecutors(_gelatoCore).stakeExecutor{value: msg.value}();
+    }
+
+    /// @dev needed for unstaking/withdrawing from GelatoCore
+    receive() external payable {
+        require(msg.sender == gelatoCore, "PermissionedExecutors.receive");
+    }
+
+    modifier onlyExecutors {
+        require(
+            msg.sender == first_executor || msg.sender == second_executor,
+            "PermissionedExecutors.onlyExecutor"
+        );
+        _;
+    }
+
+    function stakeExecutor() public payable virtual {
+        IGelatoExecutors(gelatoCore).stakeExecutor{value: msg.value}();
+    }
+
+    function unstakeExecutor() public virtual onlyExecutors {
+        uint256 stake = IGelatoProviders(gelatoCore).executorStake(address(this));
+        IGelatoExecutors(gelatoCore).unstakeExecutor();
+        msg.sender.sendValue(stake);
```

Comment:
Why dont you simple import `GelatoCore` and wrap it around `gelatoCore`? Dont have to do it with 2 different inferfaces then.

Reply:
Because I don't want to import all the GelatoCore code into the namespace. But yeah, it's a bit ugly. I will change.